ci/ansible/playbooks/os/redhat/upgrade_release.yml

```diff
@@ -0,0 +1,378 @@
+          file:
+            path: /etc/systemd/system/leapp_resume.service
+            state: absent
+
+    - name: Remove leapp packages
+      package:
+        name: "leapp*" # Leapp packages may have different nomenclature in RHEL8
+        state: absent
+
+    - name: Remove leapp configuration and log files
+      file:
+        path: /etc/leapp
+        state: absent
+
+    - name: Remove remaining RHEL 7 packages
+      block:
+        - name: Register remaining RHEL7 packages
+          shell: "rpm -qa | grep -e '\\.el7' | grep -vE '^(gpg-pubkey|libmodulemd|katello-ca-consumer)'"
+          register: remaining_rhel7_packages
+          ignore_errors: true
+
+        - name: Remove remaining RHEL7 packages
+          when: not remaining_rhel7_packages.failed
+          dnf:
+            name: "{{ remaining_rhel7_packages.stdout_lines }}"
+            state: absent
+
+        - name: Determine RHEL7 old kernel versions
+          command: "ls -1d *.el7*"
+          args:
+            chdir: /lib/modules
+          register: old_kernel_versions
+          ignore_errors: true
+
+        - name: Remove weak modules from old kernels
+          when: not old_kernel_versions.failed
+          shell: "[ -x /usr/sbin/weak-modules ] && /usr/sbin/weak-modules --remove-kernel {{ item }}"
+          loop: "{{ old_kernel_versions.stdout_lines }}"
+
+        - name: Remove old kernels from bootloader entries
+          when: not old_kernel_versions.failed
+          command: "/bin/kernel-install remove {{ item }} /lib/modules/{{ item }}/vmlinuz"
+          loop: "{{ old_kernel_versions.stdout_lines }}"
+
+    - name: Restore root login only if it was originally enabled
+      when: sshd_config.changed
+      lineinfile:
+        dest: /etc/ssh/sshd_config
+        regexp: '^PermitRootLogin no'
+        line: 'PermitRootLogin yes'
+        state: present
+
+    - *RESTART_SSHD
+
+    - name: Replace epel7 by epel8 repository
+      when:
+        - provider == "non_cloud"
+        - "'Extra Packages for Enterprise Linux 7' in pre_upgrade_active_repositories.stdout"
+      block:
+        - name: Remove epel7 repository
+          package:
+            name: epel-release
+            state: absent
+
+        - name: Install epel8 repository
+          command: "yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm"
+
+    - name: Unset Red Hat subscription manager to consume the latest RHEL content
+      when: provider == 'non_cloud'
+      command: 'subscription-manager release --unset'
+      ignore_errors: true
+
+    - *UPDATE_PACKAGES
+
+    # SELINUX configuration restore
+    - name: Re-enable SELinux
+      when:
+        - not pre_upgrade_selinux_status.stdout == "disabled"
+        - pre_upgrade_selinux_mode == "Enforcing"
+      ansible.posix.selinux:
+        state: enforcing
+
+    - name: Remove cloud-init cache # Initially populated with python2 dependencies that are unmet in RHEL8
```

Comment:
Could we do that before rebooting? Then maybe the `Remount azure specific cloud-init filesystem` or `Restart temp-disk-dataloss-warning service` tasks would be not needed.